yarl/_url.py

```diff
@@ -957,6 +958,7 @@ def _make_child(self, paths: "Sequence[str]", encoded: bool = False) -> "URL":
                     f"Appending path {path!r} starting from slash is forbidden"
                 )
             path = path if encoded else self._PATH_QUOTER(path)
+            needs_normalize |= "." in path
             segments = list(reversed(path.split("/")))
```

Comment:
This could call `.reverse()` instead and avoid making a copy

Reply:
There are a few other small cleanups that could be done here. But for another PR